audformat/core/utils.py

```diff
@@ -395,7 +393,15 @@ def difference(
     counting = collections.Counter(index)
     index = [idx for idx, count in counting.items() if count == 1]
 
-    index = _alike_index(objs[0], index)
+    # select object with highest dtype for integers
+    alike_obj = objs[0]
+    if 'int64' in _dtypes(alike_obj):
```

Comment:
What if we have two integer levels but the object that is now selected has only `Int64` in one of them, whereas another one has it in the other place?

Reply:
I wonder if we should simply always convert integer levels to `Int64` in the result of `union()`, `intersect()` and `difference()`?

Reply:
> What if we have two integer levels but the object that is now selected has only Int64 in one of them, whereas another one has it in the other place?

You mean something like this:
```python
index = pd.MultiIndex.from_arrays([[0], [1]], names=['idx1', 'idx2'])
index1 = audformat.utils.set_index_dtypes(index, {'idx1': 'Int64', 'idx2': 'int64'})
index2 = audformat.utils.set_index_dtypes(index, {'idx1': 'int64', 'idx2': 'Int64'})
audformat.utils.difference([index1, index2]).dtypes
```
There we expect that it returns
```
idx1    Int64
idx2    Int64
dtype: object
```
but what we get is
```python
idx1    int64
idx2    Int64
dtype: object
```

Reply:
yes

Reply:
We can probably fix this, but code will become quite ugly. So always converting to `Int64` makes sense I think. Also because we anyway convert to `Int64` with `Table` and `MiscTable`.

Reply:
OK, I changed to always converting to `Int64`.